clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -211,9 +211,8 @@ LValue CIRGenFunction::emitUnaryOpLValue(const UnaryOperator *e) {
     if (e->getType()->isAnyComplexType()) {
       cgm.errorNYI(e->getSourceRange(), "UnaryOp complex inc/dec");
       return LValue();
-    } else {
-      emitScalarPrePostIncDec(e, lv, isInc, /*isPre=*/true);
     }
+    emitScalarPrePostIncDec(e, lv, isInc, /*isPre=*/true);
```

Comment:
The `if` statement above this won't return once it's properly implemented. It really doesn't need to now. It should just be setting `lv` to a dummy value. Either way, I'd prefer to see this remain an `else` clause.